src/settings/settings.ts

```diff
@@ -215,6 +215,7 @@ export function getXMLSettings(javaHome: string | undefined, logfile: string, ex
   xml['xml']['logs']['file'] = logfile;
   xml['xml']['useCache'] = true;
   xml['xml']['java']['home'] = javaHome;
+  xml['xml']['validation']['resolveExternalEntities'] = workspace.getConfiguration('files').get('resolveExternalEntities', false);
```

Comment:
This reads `files.resolveExternalEntities` from the user's settings. You need to read `xml.validation.resolveExternalEntities` from the user's settings. As well, you need to set this setting to `false` if the user doesn't trust the workspace.